src/gui/src/layoutViewer.cpp

```diff
@@ -634,33 +634,56 @@ void LayoutViewer::drawTracks(dbTechLayer* layer,
     return;
   }
 
+  int min_resolution = 2*minimumViewableResolution();
+  Rect block_bounds;
+  block->getBBox()->getBox(block_bounds);
+  const Rect draw_bounds = block_bounds.intersect(bounds);
+
   bool is_horizontal = layer->getDirection() == dbTechLayerDir::HORIZONTAL;
   std::vector<int> grids;
   if ((!is_horizontal && options_->arePrefTracksVisible())
       || (is_horizontal && options_->areNonPrefTracksVisible())) {
-    grid->getGridX(grids);
-    for (int x : grids) {
-      if (x < bounds.xMin()) {
-        continue;
-      }
-      if (x > bounds.xMax()) {
-        break;
+    bool show_grid = true;
+    for (int i = 0; i < grid->getNumGridPatternsX(); i++) {
+      int origin, line_count, step;
+      grid->getGridPatternX(i, origin, line_count, step);
+      show_grid &= step > min_resolution;
```

Comment:
I'm not wed to either. I thought a "global" decision would be less confusing (also, the getGrids returns all the patterns together, so you couldn't use getGrids).
When the tracks become less than a pixel wide, you might as well just draw a box over the design. Drawing the tracks also takes a lot of time for m1 when looking at the whole chip.
If you're preference is to keep the visible tracks but they are spaced apart less than a pixel, it might be better to then draw a QRect so that doesn't impact the speed of the gui?

Reply:
We can leave it this way.